### PR TITLE
Update pre-commit configuration

### PR DESCRIPTION
**Pull request by codificat**
## Related Issues and Dependencies

- Trigger: https://github.com/thoth-station/thoth-application/issues/2111
- Includes: #71 

## This introduces a breaking change

Not really breaking, but: this PR removes the mypy pre-commit check, as the current code base has various mypy problems that would take a bit to fix.

## This Pull Request implements

- Use https for pre-commit plugin repos (ref: https://github.com/thoth-station/thoth-application/issues/2111)
- Update plugin versions
- Apply changes resulting from the updated pre-commit plugins checks

This also includes the change proposed by #71. Due to the outdated pre-commit configuration, that PR will not pass pre-commit checks.

Alternatively, we could fix pre-commit in this PR and re-base #71 afterwards, but I thought that for that 1-char change it might be simpler to just add it here...

### CODE REVIEW

File: .aicoe-ci.yaml [hunk 1]
```diff
@@ -1,5 +1,4 @@
 check:
   - thoth-pytest-py38
```
Comment (harshad16) [inline .aicoe-ci.yaml]:
```suggestion
  
```

**Review by harshad16 — CHANGES_REQUESTED** [review]
lgtm
one more change needed.

File: .aicoe-ci.yaml (hunk 1 above)
Comment (codificat) [reply .aicoe-ci.yaml]: Right, thanks!

Although the equivalent prow pre-submit is missing from the config.

So, I will add the config in `.prow.yaml` and remove the check from `.aicle-ci.yaml`

**Review by harshad16 — APPROVED** [review]
lgtm
thanks :100:

**Pull request merged**